src/components/PostDetails.tsx

```diff
@@ -1,117 +1,162 @@
-import React from 'react';
-import { Loader } from './Loader';
+import React, { useEffect, useState } from 'react';
+import { Loader } from './Loader/Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { ErrMessage } from '../types/ErrMessage';
+import { CommentsList } from './CommentsList';
+import { addComment, deleteComment, getComments } from '../api';
+
+type Props = {
+  selectedPost: Post,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+}) => {
+  const [isLoadingComments, setIsLoadingComments] = useState<boolean>(false);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [errorMess, setErrorMess] = useState<ErrMessage>(ErrMessage.Empty);
+  const [isShowingForm, setIsShowingForm] = useState<boolean>(false);
+  const [formName, setFormName] = useState<string>('');
+  const [formEmail, setFormEmail] = useState<string>('');
+  const [formText, setFormText] = useState<string>('');
+  const [
+    isLoadingAddComment, setIsLoadingAddComment,
+  ] = useState<boolean>(false);
+
+  const { id, title, body } = selectedPost;
+
+  const handlerOpeningForm = () => {
+    setIsShowingForm(true);
+    setErrorMess(ErrMessage.Empty);
+  };
+
+  const addComments = (
+    newName: string,
+    newEmail: string,
+    newText: string,
+  ) => {
+    const newComment = {
+      id: Math.max(...comments.map(c => c.id)) + 1,
```

Comment:
Generating unique id with Math.max might not work as expected if 'comments' array is empty. This will result in NaN.